### PR TITLE
Add new dynamic-forms module

### PR DESCRIPTION
**Pull request by NipuniBhagya**
### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: modules/dynamic-forms/package.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
+{
+    "private": true,
+    "name": "@wso2is/dynamic-forms",
+    "version": "0.0.1",
```
Comment (brionmario) [inline modules/dynamic-forms/package.json:4]: Lets use the same version as other modules.

File: modules/dynamic-forms/src/components/dynamic-form.scss [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+.dynamic-form {
```
Comment (brionmario) [inline modules/dynamic-forms/src/components/dynamic-form.scss:19]: Shouldn't this come from the application's side?

File: modules/dynamic-forms/src/components/dynamic-wizard-page.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * Dynamic wizard page component.
+ * 
+ * @param props - Props injected to the component.
+ */
+export const DynamicWizardPage = (props: PropsWithChildren<Record<string, any>>): ReactElement => {
+
+    const { children } = props;
+
+    return children;
+};
```
Comment (brionmario) [inline modules/dynamic-forms/src/components/dynamic-wizard-page.tsx:31]:
```suggestion
};

```

File: modules/dynamic-forms/src/components/dynamic-wizard.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        setTotalPages,
+        pageChanged,
+        initialValues,
+        uncontrolledForm,
+        ...rest
+    } = props;
+
+    // Form reference to trigger the form submit externally
+    // on each page submission.
+    const formRef: MutableRefObject<any> = useRef(null);
```
Comment (brionmario) [inline modules/dynamic-forms/src/components/dynamic-wizard.tsx:78]: Check if `HTMLFormElement` works here.
```suggestion
    const formRef: MutableRefObject<HTMLFormElement> = useRef<HTMLFormElement>(null);
```

File: modules/dynamic-forms/src/components/dynamic-wizard.tsx [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+    const handleSubmit = (values: any, form: FormApi<Record<string, any>>) => {
+        if (currentPageIndex === lastPageIndex) {
+            return onSubmit(values, form);
+        } else {
+            setValues(values);
+        }
+    };
+
+    const validate = (values: { [ key: string ]: any }) => {
+        const activePage: any = React.Children.toArray(children)[ currentPageIndex ];
```
Comment (brionmario) [inline modules/dynamic-forms/src/components/dynamic-wizard.tsx:132]:
```suggestion
        const activePage: ReactElement = React.Children.toArray(children)[ currentPageIndex ];
```

File: modules/dynamic-forms/src/components/utils.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+ */
+
+import React, { ReactElement } from "react";
+
+import { FieldMetaState, FieldProps, useField } from "react-final-form";
+import { FormHelperText, FormHelperTextProps } from "@oxygen-ui/react";
+import DynamicFormField from "./dynamic-form-field";
+
+export interface ErrorMessageProps {
+	showError: boolean;
```
Comment (brionmario) [inline modules/dynamic-forms/src/components/utils.tsx:26]: Check the indentation.

**Review by brionmario — APPROVED** [review]

File: modules/dynamic-forms/src/components/utils.tsx (hunk 6 above)
Comment (NipuniBhagya) [reply modules/dynamic-forms/src/components/utils.tsx:26]: The indentation is 1 tab space in the file as required. Seems like and issue with the way it's displayed here.

**Pull request merged**